### ISSUE TITLE
Gradle issue: Using the incremental task API without declaring any outputs has been deprecated.

### ISSUE THREAD

**Issue by ZacSweers** [diffplug/spotless#427]
Gradle 5.5.1
Spotless 3.24.0
MacOS 10.14.6
Github repo: https://github.com/zacsweers/catchup

Full trace reported from gradle below. Unfortunately it doesn't say _where_ the issue is from in code, just that it occurs when running the `spotlessKotlin` task on any project that applies it

```
Warning deprecated usage found: Using the incremental task API without declaring any outputs has been deprecated.

org.gradle.api.internal.changedetection.changes.DefaultTaskExecutionModeResolver.getExecutionMode(DefaultTaskExecutionModeResolver.java:46)
org.gradle.api.internal.tasks.execution.ResolveTaskExecutionModeExecuter.execute(ResolveTaskExecutionModeExecuter.java:67)
org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:57)
org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:56)
org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:36)
org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.executeTask(EventFiringTaskExecuter.java:77)
org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:55)
org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:52)
org.gradle.internal.operations.DefaultBuildOperationExecutor$CallableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:416)
org.gradle.internal.operations.DefaultBuildOperationExecutor$CallableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:406)
org.gradle.internal.operations.DefaultBuildOperationExecutor$1.execute(DefaultBuildOperationExecutor.java:165)
org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:250)
org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:158)
org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:102)
org.gradle.internal.operations.DelegatingBuildOperationExecutor.call(DelegatingBuildOperationExecutor.java:36)
org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter.execute(EventFiringTaskExecuter.java:52)
org.gradle.execution.plan.LocalTaskNodeExecutor.execute(LocalTaskNodeExecutor.java:43)
org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:355)
org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:343)
org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:336)
org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:322)
org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker$1.execute(DefaultPlanExecutor.java:134)
org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker$1.execute(DefaultPlanExecutor.java:129)
org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.execute(DefaultPlanExecutor.java:202)
org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.executeNextNode(DefaultPlanExecutor.java:193)
org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.run(DefaultPlanExecutor.java:129)
org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)
java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:56)
java.base/java.lang.Thread.run(Thread.java:834)
```

**Comment by nedtwigg** [comment]
Thanks for finding this!  Each `SpotlessTask` already has one synthetic output file, e.g. for java it is `build/spotlessJava`, which we write to on every run.

https://github.com/diffplug/spotless/blob/b6b7f964ff4af04ea54cd8914bb41de222ffd027/plugin-gradle/src/main/java/com/diffplug/gradle/spotless/SpotlessTask.java#L119-L129

We could easily list this file as an output, but the trick is we also need to list it as an input (it fixes a tricky up-to-date corner-case, [described here](https://discuss.gradle.org/t/two-cases-where-the-incremental-api-is-lacking/24266)).  If having this file as both an input and an output triggers their warnings (which would be reasonable), then we can just make another synthetic output and list that as our output.

Until these deprecated warnings become errors, this won't be high on my todo list.  Happy to merge a PR before then, or I'll get to it when it becomes urgent.

**Comment by ZacSweers** [comment]
I'll see what I can do as a PR 👍. They're not errors, but they do warn on every build in gradle 5.x+

**Comment by nedtwigg** [comment]
Released in 3.24.2
